common/evmcompat/helpers.go

```diff
@@ -80,7 +83,7 @@ func GenerateTypedSig(data []byte, privKey *ecdsa.PrivateKey, sigType SignatureT
 func RecoverAddressFromTypedSig(hash []byte, sig []byte) (common.Address, error) {
 	var signer common.Address
 
-	if len(sig) != 66 {
+	if len(sig) != 66 && SignatureType(sig[0]) != SignatureType_EOS {
```

Comment:
What's the expected signature length for EOS? Check that.

Reply:
It's checked inside the switch `recoverAddressFromEosSig`
```go
func recoverAddressFromEosSig(hash []byte, sig []byte) (common.Address, error) {
	var signer common.Address
	if len(sig) != 67 {
		return signer, fmt.Errorf("eos signature must be 67 bytes, not %d bytes", len(sig))
	}
```
I could move all length checks inside the switch or put all the length in the same place.

Reply:
Split out the Geth & Trezor recovery into separate functions, and put the length checks in there.